### PR TITLE
Added Activation,padding and stride scenarios.

### PR DESCRIPTION
**Pull request by amitsrivastava78**
This is one of the TODO in the file.

### CODE REVIEW

File: tensorflow/lite/kernels/depthwise_conv_test.cc [hunk 1]
```diff
@@ -39,8 +39,73 @@ using ::testing::ElementsAreArray;
...
+  BaseDepthwiseConvolutionOpModel(
+      TfLiteRegistration* registration, const TensorData& input,
+      const TensorData& filter, const TensorData& output, Padding padding_type,
+      int stride_width, int stride_height,
+      const ActivationFunctionType& fused_activation_function,
+      int dilation_factor = 1) {
+    input_ = AddInput(input);
+    filter_ = AddInput(filter);
+
+    int bias_size = GetShape(filter_)[3];
```
Comment (shahzadlone) [inline tensorflow/lite/kernels/depthwise_conv_test.cc]: could this perhaps be ```const```

File: tensorflow/lite/kernels/depthwise_conv_test.cc [hunk 2]
```diff
@@ -39,8 +39,73 @@ using ::testing::ElementsAreArray;
...
+      // This is a quantized version. The scale of 'bias' depends on the scales
+      // of input and filter. Supposedly this is correctly set during quantized
+      // training.
+      if (filter.per_channel_quantization) {
+        // per channel quantization.
+        std::vector<float> bias_scale(
+            filter.per_channel_quantization_scales.size());
+        std::vector<int64_t> bias_zero_points(
+            filter.per_channel_quantization_scales.size());
+        for (int i = 0; i < filter.per_channel_quantization_scales.size();
```
Comment (shahzadlone) [inline tensorflow/lite/kernels/depthwise_conv_test.cc]: Might be worthwhile to save ```filter.per_channel_quantization_scales.size()``` in a ```const```` variable to save the function calls and avoid code repetition.

File: tensorflow/lite/kernels/depthwise_conv_test.cc [hunk 3]
```diff
@@ -39,8 +39,73 @@ using ::testing::ElementsAreArray;
...
+                        /*scale=*/0,
+                        /*zero_point=*/0,
+                        true,
+                        /*per_channel_scale=*/bias_scale,
+                        /*per_channel_zero_point=*/bias_zero_points,
+                        /*channel_index==*/0};
+        bias_ = AddInput(bias);
+      } else {
+        // per tensor quantization.
+        auto bias_scale = GetScale(input_) * GetScale(filter_);
```
Comment (shahzadlone) [inline tensorflow/lite/kernels/depthwise_conv_test.cc]: ```const```?

File: tensorflow/lite/kernels/depthwise_conv_test.cc [hunk 4]
```diff
@@ -39,8 +39,73 @@ using ::testing::ElementsAreArray;
...
+        TensorData bias{TensorType_INT32, {bias_size}, 0, 0, bias_scale};
+        bias_ = AddInput(bias);
+      }
+    }
+
+    output_ = AddOutput(output);
+
+    int input_depth = GetShape(input_)[3];
+    int output_depth = GetShape(filter_)[3];
+    int depth_mul = output_depth / input_depth;
```
Comment (shahzadlone) [inline tensorflow/lite/kernels/depthwise_conv_test.cc]: Could these be ```const``` as well?

File: tensorflow/lite/kernels/depthwise_conv_test.cc (hunk 1 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/depthwise_conv_test.cc]: @shahzadlone , thanks for the review, i have updated the code as per your suggestion, kindly check and approve.

Regards
Amit

File: tensorflow/lite/kernels/depthwise_conv_test.cc (hunk 3 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/depthwise_conv_test.cc]: @shahzadlone , thanks for the review, i have updated the code as per your suggestion, kindly check and approve.

Regards
Amit

File: tensorflow/lite/kernels/depthwise_conv_test.cc (hunk 4 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/depthwise_conv_test.cc]: @shahzadlone , thanks for the review, i have updated the code as per your suggestion, kindly check and approve.

Regards
Amit

File: tensorflow/lite/kernels/depthwise_conv_test.cc (hunk 2 above)
Comment (amitsrivastava78) [reply tensorflow/lite/kernels/depthwise_conv_test.cc]: @shahzadlone , thanks for the review, i have updated the code as per your suggestion, kindly check and approve.

Regards
Amit

**Comment by amitsrivastava78** [comment]
@shahzadlone , thanks for the review, i have updated the code as per your suggestion, kindly check and approve.

Regards
Amit

**Review by shahzadlone — APPROVED** [review]

**Comment by amitsrivastava78** [comment]
@suharshs, can you please review the PR.

Regards 
Amit

**Comment by amitsrivastava78** [comment]
@rthadur this PR is approved by @shahzadlone , can you please change the status to Approved.

Regards
Amit

**Comment by rthadur** [comment]
> @rthadur this PR is approved by @shahzadlone , can you please change the status to Approved.
> 
> Regards
> Amit

@suharshs gentle ping to approve this PR, @amitsrivastava78 thanks for your contribution, waiting for a googler to approve this changes.

**Comment by amitsrivastava78** [comment]
Closing this PR since similar is already approved

Regards
Amit

**Pull request closed without merging**